blog/_posts/2020-11-08-ctlptl.md

```diff
@@ -0,0 +1,73 @@
+---
+slug: "ctlptl"
+date: 2020-11-08
+author: nick
+layout: blog
+title: "Announcing ctlptl: Mess Around with Local Kubernetes without Consequences"
+image: "/assets/images/ctlptl/joe-cox-HVXujL72Dug-unsplash.jpg"
+image_caption: "Photo by <a href=\"https://unsplash.com/@joecoxx?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText\">Joe Cox</a> on <a href=\"https://unsplash.com/s/photos/ducks-puddle?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText\">Unsplash</a></span>"
+tags:
+  - kubernetes
+  - local development
+---
+
+At Tilt, we want to make Kubernetes an approachable environment for local dev.
+
+We found ourselves spending a lot of time helping teams debug misconfigurations
+with their local cluster.
+
+We wrote docs like [Choosing a local dev
+cluster](https://docs.tilt.dev/choosing_clusters.html) and example repos like
+[kind-local](https://github.com/tilt-dev/kind-local),
+[minikube-local](https://github.com/tilt-dev/minikube-local), and
+[k3d-local](https://github.com/tilt-dev/k3d-local) to help people get set up.
+
+`ctlptl` is a culmination of what we've learned.
+
+## What is `ctlptl`?
+
+[`ctlptl`](https://github.com/tilt-dev/ctlptl) is a CLI for declaratively setting up local Kubernetes clusters.
+
+Inspired by `kubectl` and
+[ClusterAPI's](https://github.com/kubernetes-sigs/cluster-api) `clusterctl`, you
+declare your local cluster with YAML and use `ctlptl` to set it up.
+
+## How should I pronounce `ctlptl`?
+
+We've talked a lot about this. We're not any closer to a single answer. We've
+decided that these are all acceptable pronounciations:
+
+- Cattle paddle
+- Control pedal
+- Cartel patrol
+- Cuddle petal
+- Cuttle puddle
+
+## Who is `ctlptl` for?
+
+`ctlptl` is a building block for platform teams setting up a multi-service dev environments locally.
+
+To have a consistent environment across machines, you need to be able to stitch together:
+
+- a dev cluster
+- a dev registry
+- storage
+- mounts
+- cpu
+- memory
+
+If one of these is off (e.g., one dev accidentally set their CPU too low), the
+environment is awful and uncomfortable.
+
+A well-organized `ctlptl` will help to fix that.
```

Comment:
Maybe worth a line for:

> `ctlptl` helps a single dev easily set up their dev environment according to best practices; and it gives a Tilt maintainer a one-liner to give their team for easy cluster setup.

Or something like that--just to explicitly call out the diff personas.

Reply:
i like it! done